### PR TITLE
Travis codecov

### LINKED ISSUE #306: Integrate with codecov.io
The goal is to get coverage reports on every PR, to get coverage badges which update themselves and remove the old infrastructure for manually generating and displaying coverage badges, and finally to shore up areas of poor test coverage.

### PR DESCRIPTION
**Pull request by JacobMGEvans**
will close #306 when complete.

### CODE REVIEW

**Comment by JacobMGEvans** [comment]
Hmmm, not seeing any Travis scripts running... I am going to keep this PR open for testing CI pushes

**Comment by JacobMGEvans** [comment]
Actually I am going to close and move it to a local PR from my fork

**Comment by conartist6** [comment]
Cool. I think there should be a coverage script (`npm run coverage`), and you just have the `verify` script run it. `verify` is meant be "the script that CI runs". Also hopefully you won't need to run any script to generate the badges anymore, you can use the codecov one from [here](https://shields.io/category/coverage)

**Comment by JacobMGEvans** [comment]
> Cool. I think there should be a coverage script (`npm run coverage`), and you just have the `verify` script run it. `verify` is meant be "the script that CI runs". Also hopefully you won't need to run any script to generate the badges anymore, you can use the codecov one from [here](https://shields.io/category/coverage).
```yaml

  script:
  - npm run verify
  - npm run coverage

```

**Comment by JacobMGEvans** [comment]
This badge seems to not be able to access the repo for some reason... 
https://img.shields.io/codecov/c/github/iter-tools/iter-tools

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 language: node_js
 node_js:
-  - "10"
+  - '10'
 
-script: npm run verify
+
+  script:
```
Comment (conartist6) [inline .travis.yml]: Your yaml is unparsable because script isn't at the beginning of the line here. That's why Travis isn't running.

Comment (JacobMGEvans) [reply .travis.yml]: Hmm, this was valid when I looked at their docs, wonder why it's not working here
:(

Comment (JacobMGEvans) [reply .travis.yml]: Oh I see you want only one CI script to run

Comment (conartist6) [reply .travis.yml]: Yes. It's not a travis problem, your yaml syntax is just unparseable. YAML is indentation sensitive. Try pasting what you've written into, say, https://yaml-online-parser.appspot.com/ and you'll see what I mean

File: package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     "build:es2018-cjs": "cross-env BABEL_ENV=cjs babel --config-file ./babel-es2018.config.js -x .mjs src -d es2018",
     "build:copyts": "copyfiles -f src/*.ts es5 && copyfiles -f src/*.ts es2015 && copyfiles -f src/*.ts es2018",
     "create:method": "node ./generate-yo method",
-    "verify": "run-s generate verify:* build test",
+    "verify": "run-s generate verify:* build test && npm run coverage",
```
Comment (conartist6) [inline package.json]: This is a [npm-run-all](https://www.npmjs.com/package/npm-run-all) command, it only takes patterns for the names of other scripts. This would just be `build test coverage`, except that coverage is running the tests, so just `build coverage`.

File: .travis.yml (hunk 1 above)
Comment (JacobMGEvans) [reply .travis.yml]: I think my change fixes this. Thank you :)

File: package.json (hunk 2 above)
Comment (JacobMGEvans) [reply package.json]: Understood thanks for the explanation.

Comment (JacobMGEvans) [reply package.json]: is that `test:ts` or `test:js `?

File: .travis.yml (hunk 1 above)
Comment (JacobMGEvans) [reply .travis.yml]: Oops lol I used Groovy with Jenkins so this is a bit different than what I am use to lol

File: package.json (hunk 2 above)
Comment (conartist6) [reply package.json]: `test:ts`, because coverage is a form of `test:js`. Actually I think to make that abundantly clear the `coverage` script should be renamed to `test:js:coverage`

File: README.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ Iter-tools is at present the only fully-featured library of its kind. Here is wh
 - **Pay what you need on web**: Iter-tools is designed to be modular, meaning that only the methods you actually use end up being bundled and shipped if you are using a supported bundler. Webpack/uglify are capable of tree-shaking iter-tools, as is Rollup after v1.3.0. If you are not using a bundler that supports tree-shaking, you can still transpile an expression like `import { map, filter } from "iter-tools"` into imports of individual files: `iter-tools/es5/map` and `iter-tools/es5/filter`. To do this use the `iter-tools-explode-imports` package.
 
 ## Definitions
-This should help clarify the documentation. You can also get more informations here: https://developer.mozilla.org/en/docs/Web/JavaScript/Guide/Iterators_and_Generators and here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols
```
Comment (conartist6) [inline README.md]: Why are you deleting these documentation links? Those are definitely supposed to be there

Comment (JacobMGEvans) [reply README.md]: That was a mistake my bad :( 

Sorry I wasn't paying attention...

File: coverage/badge-branches.svg [hunk 4]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="170" height="20"><linearGradient id="b" x2="0" y2="100%"><stop offset="0" stop-color="#bbb" stop-opacity=".1"/><stop offset="1" stop-opacity=".1"/></linearGradient><clipPath id="a"><rect width="170" height="20" rx="3" fill="#fff"/></clipPath><g clip-path="url(#a)"><path fill="#555" d="M0 0h117v20H0z"/><path fill="#dfb317" d="M117 0h53v20H117z"/><path fill="url(#b)" d="M0 0h170v20H0z"/></g><g fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="110"> <text x="595" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="1070">Coverage:branches</text><text x="595" y="140" transform="scale(.1)" textLength="1070">Coverage:branches</text><text x="1425" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="430">85.53%</text><text x="1425" y="140" transform="scale(.1)" textLength="430">85.53%</text></g> </svg>
```
Comment (conartist6) [inline coverage/badge-branches.svg]: These badges shouldn't be updated, they should be deleted.

File: .travis.yml [hunk 5]
```diff
@@ -1,5 +1,7 @@
 language: node_js
 node_js:
-  - "10"
+  - '10'
+
+
+  script: npm run verify
```
Comment (conartist6) [inline .travis.yml]: It seems there are no longer any differences in this file, so lets just revert the changes here.

File: coverage/badge-branches.svg (hunk 4 above)
Comment (JacobMGEvans) [reply coverage/badge-branches.svg]: I thought you wanted a dynamic badge from codecov?

Comment (JacobMGEvans) [reply coverage/badge-branches.svg]: Oh nvm, yeah I was trying something, in the beginning, these will be removed with the other wiring.

File: .travis.yml (hunk 5 above)
Comment (JacobMGEvans) [reply .travis.yml]: Done.
